### DISCUSSION TITLE
着地位置修正や着地時間修正で使用する着地までの残り時間

### DISCUSSION THREAD

**Discussion by Naoki-Hiraoka** [YutaKojio/hrpsys-base#60] [Q&A] [answered]
着地位置修正や着地時間修正で使用する着地までの残り時間として、次の両脚支持期の終了時までの時間(`remain_count`)を用いていますが、現在の遊脚が着地するまでの時間を用いないのはなぜでしょうか。

https://github.com/YutaKojio/hrpsys-base/blob/49aeca84efde2d52655e3ee6b0fb0fc1219861cb/rtc/AutoBalancer/GaitGenerator.cpp#L1689
https://github.com/YutaKojio/hrpsys-base/blob/49aeca84efde2d52655e3ee6b0fb0fc1219861cb/rtc/AutoBalancer/GaitGenerator.cpp#L1758

**Comment by YutaKojio** [comment, marked as answer]
num_preview_step = 1の場合，
重心軌道生成(update_foot_guided_controller)では，remain_count = 0時点でCPと着地位置が一致するように制御しています．
この重心軌道生成において，ZMPが支持足平制約を満たす範囲で軌道を生成できないような場合に，着地位置・時間修正が行われていて，
足平制限を満たす範囲で終端制約を満たせるように着地位置・時間修正量の計算をすることになるので，用いる残り時間はもちろん同じ残り時間になります．

ただし， 現在は num_preview_step = 2にしているので，着地位置・時間修正は本当は過剰にやりすぎていることになっています．
ので，着地位置・時間修正（visionありの場合も）でn-stepを考慮するのはTODOとしてあります．

**Reply by YutaKojio** [reply]
理論的に正しくはないけど，バグというわけでもないので，enhancementラベルにしておきます
